src/Command/PopulateCommand.php

```diff
@@ -220,35 +213,33 @@ private function populateIndexType(OutputInterface $output, $index, $type, $rese
             $this->resetter->resetIndexType($index, $type);
         }
 
-        $offset = $options['offset'];
+        $offset = 1 < $options['first_page'] ? ($options['first_page'] - 1) * $options['max_per_page'] : 0;
         $loggerClosure = ProgressClosureBuilder::build($output, 'Populating', $index, $type, $offset);
 
-        if ($loggerClosure) {
-            $this->dispatcher->addListener(
-                Events::ON_EXCEPTION,
-                function(OnExceptionEvent $event) use ($loggerClosure, $options) {
-                    $loggerClosure(
-                        $options['batch_size'],
-                        count($event->getObjects()),
-                        sprintf('<error>%s</error>', $event->getException()->getMessage())
-                    );
-                }
-            );
+        $this->dispatcher->addListener(
+            Events::ON_EXCEPTION,
+            function(OnExceptionEvent $event) use ($loggerClosure) {
+                $loggerClosure(
+                    count($event->getObjects()),
+                    $event->getPager()->getNbResults(),
+                    sprintf('<error>%s</error>', $event->getException()->getMessage())
+                );
+            }
+        );
 
-            $this->dispatcher->addListener(
-                Events::POST_INSERT_OBJECTS,
-                function(PostInsertObjectsEvent $event) use ($loggerClosure) {
-                    $loggerClosure(count($event->getObjects()), $event->getPager()->getNbResults());
-                }
-            );
+        $this->dispatcher->addListener(
+            Events::POST_INSERT_OBJECTS,
+            function(PostInsertObjectsEvent $event) use ($loggerClosure) {
+                $loggerClosure(count($event->getObjects()), $event->getPager()->getNbResults());
+            }
+        );
 
-            $this->dispatcher->addListener(
-                Events::POST_ASYNC_INSERT_OBJECTS,
```

Comment:
@makasim that event does not seem to be dispatched at all. What's the idea of that?

Reply:
It should be used by async pager persister. It does not deal with objects directly but schedule sub jobs to some workers. Like here https://github.com/php-enqueue/enqueue-elastica-bundle/blob/master/Persister/QueuePagerPersister.php#L125